Resources/Private/JavaScript/media-module/src/components/TopBar/SortOrderSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { useCallback, useMemo } from 'react';
+import { useRecoilState } from 'recoil';
+
+import { SelectBox, IconButton } from '@neos-project/react-ui-components';
+
+import { createUseMediaUiStyles, useIntl, useMediaUi } from '@media-ui/core/src';
+
+import selectedSortOrderState, { SORT_BY, SORT_DIRECTION } from '@media-ui/core/src/state/selectedSortOrderState';
+
+const useStyles = createUseMediaUiStyles({
+    sortingState: {
+        display: 'flex',
+        flex: 1,
+        minWidth: 0,
+    },
+    selectBox: {
+        minWidth: 0,
+    },
+});
+
+interface SortByOption {
+    value: SORT_BY;
+    label: string;
+    icon: string;
+}
+
+const SortOrderSelector: React.FC = () => {
+    const { isInNodeCreationDialog, selectionMode } = useMediaUi();
+    const classes = useStyles();
+    const [sortOrderState, setSortOrderState] = useRecoilState(selectedSortOrderState);
+    const { translate } = useIntl();
+    const hideOptionIcon = isInNodeCreationDialog || selectionMode;
+    const handleChangeSortBy = useCallback(
+        (sortBy: SORT_BY) => {
+            setSortOrderState({ ...sortOrderState, sortBy });
+        },
+        [sortOrderState, setSortOrderState]
+    );
+
+    const handleSwitchSortDirection = useCallback(() => {
+        setSortOrderState({
+            ...sortOrderState,
+            sortDirection:
+                sortOrderState.sortDirection === SORT_DIRECTION.Asc ? SORT_DIRECTION.Desc : SORT_DIRECTION.Asc,
+        });
+    }, [sortOrderState, setSortOrderState]);
+
+    const sortByOptions: SortByOption[] = useMemo(() => {
+        return [
+            {
+                value: SORT_BY.LastModified,
+                label: translate('sortingState.sortBy.values.lastModified', 'Last Modified'),
+                icon: hideOptionIcon ? '' : 'calendar',
+            },
+            {
+                value: SORT_BY.Name,
+                label: translate('sortingState.sortBy.values.name', 'Name'),
+                icon: hideOptionIcon ? '' : 'font',
+            },
+        ];
+    }, [translate]);
+
+    return (
+        <div className={classes.sortingState}>
+            <div className={classes.selectBox}>
```

Comment:
The selectbox itself doesn't seem to respect classnames

Reply:
Hm it should. The component supports it. Maybe a JSS related issue. Will check.

Reply:
OK the `className` is just added to the inner div. Your solution is fine for now. To do it proper, one would have to use the `themr` api.